test-data/unit/check-classes.test

```diff
@@ -1060,6 +1060,16 @@ def f() -> None:
     a.g(a) # E: Too many arguments for "g" of "A"
 [targets __main__, __main__.f]
 
+[case testGenericClassWithinFunction]
+from typing import TypeVar
+
+def test() -> None:
+    T = TypeVar('T', bound='Foo')
+    class Foo:
+        def bar(self, foo: T) -> T:
+            pass
```

Comment:
What about using `T` in the function body (e.g. `x: T = foo`, `reveal_type(x)`), just in case?

Would it make sense to test using `T` in the class body -- this should generate an error, since the binding is out of scope. This should probably be a separate test case (or at least a separate function in this test case) to avoid interfering with the current test case.